org.eclipse.m2e.pde.ui/src/org/eclipse/m2e/pde/ui/target/editor/internal/TargetDependencyModel.java

```diff
@@ -0,0 +1,339 @@
+		return List.copyOf(new ArrayList<>(targetDependencies));
+	}
+
+	public OperationHistoryFacade getHistory() {
+		return history;
+	}
+
+	public void setHeightHint(int heightHint) {
+		this.heightHint.setValue(heightHint);
+	}
+
+	public int getHeightHint() {
+		return heightHint.getValue();
+	}
+
+	/**
+	 * Modifies the property of the selected element. For example the group id or
+	 * version. Does nothing of the old and new value are identical. An error is
+	 * logged if the element is not selected.<br>
+	 * This operation is undoable.
+	 * 
+	 * @param source   The original, unmodified dependency.
+	 * @param setter   The accessor used to apply the given values. Must not be
+	 *                 null.
+	 * @param oldValue The original value. Restored after an {@code undo} operation.
+	 *                 May be null.
+	 * @param newValue The new value applied after an {@code execute} or
+	 *                 {@code redo} operation. May be null.
+	 */
+	public void modify(MavenTargetDependency source, BiConsumer<MavenTargetDependency, String> setter, String oldValue,
+			String newValue) {
+		if (Objects.equals(oldValue, newValue)) {
+			return;
+		}
+
+		history.propertyChange(value -> currentSelection.modify(source, setter, value), oldValue, newValue);
+	}
+
+	/**
+	 * Adds the current clipboard dependencies to the list of target dependencies.
+	 * If the clipboard is empty its content doesn't contain any valid Maven
+	 * coordinates, a blank dependency is added. The newly added elements are
+	 * selected.<br>
+	 * This operation is undoable.
+	 */
+	public void add() {
+		List<MavenTargetDependency> newCurrentSelection = getDefaultClipboardDependencies();
+		List<MavenTargetDependency> newTargetDependencies = new ArrayList<>();
+		newTargetDependencies.addAll(deepClone(getTargetDependencies()));
+		newTargetDependencies.addAll(newCurrentSelection);
+
+		history.modelChange(newTargetDependencies, newCurrentSelection);
+	}
+
+	/**
+	 * Removes the current selection from the list of target dependencies. The new
+	 * selection is calculated using the following rules:
+	 * <ul>
+	 * <li>If one element has been removed, the element at the same position (in the
+	 * new list) is selected. If this position is out-of-bounds, the last element in
+	 * the list is selected instead.</li>
+	 * <li>If the more than one element is removed or if the target list is empty,
+	 * the selection is cleared.</li>
+	 * </ul>
+	 * An error is logged if {@link canRemove()} returns {@code false}.<br>
+	 * This operation is undoable.
+	 * 
+	 * @see #canRemove()
+	 */
+	public void remove() {
+		if (!canRemove()) {
+			LOGGER.error("The remove operation is currently not possible! Ignore...");
+			return;
+		}
+
+		List<MavenTargetDependency> oldCurrentSelection = getCurrentSelection();
+		List<MavenTargetDependency> oldTargetDependencies = getTargetDependencies();
+
+		List<MavenTargetDependency> newCurrentSelection = new ArrayList<>();
```

Comment:
```suggestion
		List<MavenTargetDependency> newCurrentSelection;
```
Assigned value is not used.
If you want you could consider to initialize the variable with an empty list, reverse the condition of the if-block and remove the then empty else block.